contributors/design-proposals/device-plugin.md

```diff
@@ -0,0 +1,212 @@
+# Device Manager Proposal
+
+  * [Abstract](#abstract)
+  * [Motivation](#motivation)
+  * [Use Cases](#use-cases)
+  * [Objectives](#objectives)
+  * [Proposed Changes](#proposed-changes)
+
+
+_Authors:_
+
+* @RenaudWasTaken - Renaud Gaubert &lt;rgaubert@nvidia.com&gt;
+
+## Abstract
+
+This documents describes a solution to discovering, monitoring and representing
+external devices such as:
+  * GPUs
+  * NICs
+  * FPGAs
+  * InfiniBand
+  * ...
+
+## Motivation
+
+Kubernetes currently supports discovery of CPU and Memory primarily to a
+minimal extent. Very few devices are handled natively by Kubelet.
+
+It is not a sustainable solution to expect every vendor to add their vendor
+specific code inside kubernetes. This approach does not scale and is not portable.
+
+We want a solution for those vendors to be able to advertise their resources to the kubelet
+and monitor them.
+We also want a way for the user to specify which resource their jobs will use and what
+constraints are associated to these resources.
+
+In order to solve this problem it is obvious that we need a plugin system in
+order to have vendors advertise and monitor their resources on behalf of Kubelet.
+
+Additionally, we introduce the concept of ResourceType to be able to select
```

Comment:
I'd recommend keeping this proposal just about how kubelet will interact with third party hardware devices. This itself is an API. Let's deal with the k8s APIs as part of resource classes to avoid a deluge of comments in this proposal.